File: Python/Problem007.py
from time import time
from itertools import islice,count
from math import floor,sqrt


def prime(n):
    return n > 1 and all(n % i for i in islice(count(2), int(sqrt(n) - 1)))

start = time()
pLis, i = [], 2

while len(pLis) < 10001:
    if prime(i):
        pLis.append(i)
    i += 1
    
ans = pLis[-1]

if __name__ == '__main__':
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")

File: Go/Solutions/projeuler/Problem007.go
package euler

import (
	"fmt"
	"time"
)

/*
Problem007 answers the problem at : https://projecteuler.net/problem=7
	* Problem 7:
		By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
		What is the 10001st prime number?

*/
func Problem007() {

	start := time.Now()
	ans := PrimeSieve(1000000)[10000]
	end := time.Now()

	fmt.Printf("\nAnswer to Problem 7 : %d\n", ans)
	fmt.Printf("Time Taken: %f seconds\n\n", end.Sub(start).Seconds())
}

//PrimeSieve implements the sieve of Eratosthenes
func PrimeSieve(n int) []int {
	integers := make([]bool, n+1)
	for i := 2; i < n+1; i++ {
		integers[i] = true
	}

	for p := 2; p*p <= n; p++ {
		if integers[p] == true {
			for i := p * 2; i <= n; i += p {
				integers[i] = false
			}
		}
	}

	var primes []int
	for p := 2; p <= n; p++ {
		if integers[p] == true {
			primes = append(primes, p)
		}
	}
	return primes
}
